### ISSUE TITLE
Miniconda hook not found

### ISSUE THREAD

**Issue by Kostazs** [oobabooga/text-generation-webui#1351]
### Describe the bug

When I try to click one of the files from the extracted oobagooga folder I am hit with the error shown below.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

I clicked the "one-click installers" oobagooba_windows.zip I instaled the folder in my D drive where I install most of my folders. I extract the folder and notice that I have some missing folders according to the youtube tutorial that im watching https://youtu.be/FOyqcETVUCs?t=112 (linked with timestamp) but I brush it off and continue following along. When I try to run any folder. I get resulted with that error message, i tried searching for a solution but I found nothing and it has made me to think that the solution is probably something I haven't installed that everyone has or just installed.

### Screenshot

![error](https://user-images.githubusercontent.com/128280634/232885206-10c0c0a0-97b5-4ff1-944b-3522bf57e9bc.png)
![error 2](https://user-images.githubusercontent.com/128280634/232886395-dde255e2-aff8-408d-92da-c0a07a3eaf67.png)

### Logs

```shell
The system cannot find the path specified.

Miniconda hook not found.
Press any key to continue . . .
```

### System Info

```shell
Windows 10
64 bit
GT 1030 2GB VRAM
RYZEN 3 1200 quad-core
16gb of ram 2500mhz
```

**Comment by jllllll** [comment]
The installer was overhauled yesterday for cross-platform support. Any tutorials written before that are now out-dated. Run `start-windows.bat` to install and start the webui. I'm assuming you were trying to run the update script?

**Comment by xNul** [comment]
> I had to debug this last night - if you still see this error, it is likely that environment variable TMP is incorrectly set (it needs to match install drive, and folder needs to exist - this fixed install for me)

@desva this line of code fixed it for you?
```batch
set TMP=D:
```

**Comment by Kostazs** [comment]
> I had to debug this last night - if you still see this error, it is likely that environment variable TMP is incorrectly set (it needs to match install drive, and folder needs to exist - this fixed install for me)

Thank you bro! this fixed my problem.

**Comment by Kostazs** [comment]
> > I had to debug this last night - if you still see this error, it is likely that environment variable TMP is incorrectly set (it needs to match install drive, and folder needs to exist - this fixed install for me)
> 
> @desva this line of code fixed it for you?
> 
> ```batchfile
> set TMP=D:\temp
> ```

i changed the tmp file directory like this from this video https://youtu.be/JQfPSj7yYJw
go to Control Panel->System->Advanced, and enter the "Environment Variables" dialog window to find and change the TEMP and TMP variable settings in the "User variables".

**Comment by desva** [comment]
Yes - just need to set TMP in the batch file to match the install drive (and ensure folder exists).

**Comment by xNul** [comment]
@desva interesting, I'm not able to reproduce on my Windows system on my D drive. Could you confirm these changes fix the installation process for you? https://github.com/xNul/one-click-installers

**Comment by johkhknlknkjhk** [comment]
Qucik question can someone tell me what EXACTLY to write on the TEMP and TMP ?

**Comment by desva** [comment]
You need to set the TMP environment variable to point to a temporary folder on the same drive as your installation. Sent from my iPhone, please excuse any errors. On 24 Apr 2023, at 04:46, johkhknlknkjhk ***@***.***> wrote:﻿
Qucik question can someone tell me what EXACTLY to write on the TEMP and TMP ?

**Comment by xNul** [comment]
@johkhknlknkjhk did the changes I made work for you?

**Comment by kodra-dev** [comment]
Still doesn't work for me. Got 

```
The system cannot find the path specified.

Miniconda hook not found.
Press any key to continue . . .
```

Even after I set `TMP=C:\temp`

**Comment by xNul** [comment]
@kodra-dev I'm going to need some more info like what OS are you running, what's your install log, and what drive are you using?

**Comment by Webslug** [comment]
Yeah this definitely no longer works and I am getting the same error - It is clear that this no longer works.

Mine is installed in a folder called AI with no spaces, I have made the TEMP folder and I still get the error.  I've tried all the solutions I can think of and I'm about to give up.

Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00008fb0 (most recent call first):
  <no Python frame>

Miniconda hook not found.

**Comment by jllllll** [comment]
> Yeah this definitely no longer works and I am getting the same error - It is clear that this no longer works.
> 
> Mine is installed in a folder called AI with no spaces, I have made the TEMP folder and I still get the error. I've tried all the solutions I can think of and I'm about to give up.
> 
> Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding Python runtime state: core initialized ModuleNotFoundError: No module named 'encodings'
> 
> Current thread 0x00008fb0 (most recent call first):
> 
> Miniconda hook not found.

Try adding this to `start_windows.bat` just after `set TEMP=%cd%\installer_files`:
```cmd
SET PYTHONNOUSERSITE=1
SET PYTHONPATH=
```

**Comment by Webslug** [comment]
Thanks Jlllll, I added the two lines which you suggested however I still get the error, can you help further?



**SET PYTHONNOUSERSITE=1
SET PYTHONPATH=C:\Program Files\Python310**

Python path configuration:
  PYTHONHOME = 'C:\Program Files\Python310\Scripts'
  PYTHONPATH = 'C:\Program Files\Python310'
  program name = 'D:\AI\oobabooga\installer_files\conda\python.exe'
  isolated = 0
  environment = 1
  user site = 0
  import site = 1
  sys._base_executable = 'D:\\AI\\oobabooga\\installer_files\\conda\\python.exe'
  sys.base_prefix = 'C:\\Program Files\\Python310\\Scripts'
  sys.base_exec_prefix = 'C:\\Program Files\\Python310\\Scripts'
  sys.platlibdir = 'lib'
  sys.executable = 'D:\\AI\\oobabooga\\installer_files\\conda\\python.exe'
  sys.prefix = 'C:\\Program Files\\Python310\\Scripts'
  sys.exec_prefix = 'C:\\Program Files\\Python310\\Scripts'
  sys.path = [
    'C:\\Program Files\\Python310',
    'D:\\AI\\oobabooga\\installer_files\\conda\\python310.zip',
    'C:\\Program Files\\Python310\\Scripts\\DLLs',
    'C:\\Program Files\\Python310\\Scripts\\lib',
    'D:\\AI\\oobabooga\\installer_files\\conda',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00004090 (most recent call first):
  <no Python frame>

Miniconda hook not found.

**Comment by Webslug** [comment]
Numerous people on Twitter also mentioned that the new installer is broken and they also could not get it to install.

**Comment by jllllll** [comment]
@Webslug Why did you add a path to PYTHONPATH? It was left deliberately blank to clear it. Try it again with PYTHONHOME as well:
```cmd
SET PYTHONNOUSERSITE=1
SET PYTHONPATH=
SET PYTHONHOME=
```
PYTHONPATH and PYTHONHOME should never be set to anything. I replicated this exact error by setting those variables to a path. They need to be unset.

**Comment by Webslug** [comment]
Thanks [jllllll](https://github.com/jllllll) that seems to have solved the problem.

**Comment by jllllll** [comment]
> Thanks [jllllll](https://github.com/jllllll) that seems to have solved the problem.
> 
> After I selected A for Nvidia, the script downloads a bunch of files, however after 5 minutes, it just seems to stop and I am worried it has crashed. Am I just being impatient?

You'll know if it crashed. The script uses Miniconda to download things, which has a fairly small limit on the amount of downloads it shows. Sometimes, for unknown reasons, it will take longer than usual and seem like it has frozen.

**Comment by SuperFurias** [comment]
I had the same issue, all my files inside the folder started showing up the error "Miniconda hook not found" and "The system cannot find the path specified"

...after some time.. i tried manually installing conda inside my folder.
...it seems that conda is unable to install in folders that start with "!" inside.
..my folder was called "!TextGenerationGUI" because i wanted it to always stay on top of other folders..
..but well.. conda is unable to install in folders with !.
..now, the problem is solved.

**Comment by SuperFurias** [comment]
> I had the same issue, all my files inside the folder started showing up the error "Miniconda hook not found" and "The system cannot find the path specified"
> 
> ...after some time.. i tried manually installing miniconda inside my folder. by opening the file that the bat file automatically downloaded.. and... well.. ...it seems that miniconda is unable to install in folders that start with "!" inside. ..my folder was called "!TextGenerationGUI" because i wanted it to always stay on top of other folders.. ..but well.. miniconda is unable to install in folders with "!". so i had rename my folder. ..now, after removing the "!" from the folder name, the problem seems to be solved.

I think that the installer should warn users that it is impossible to run/install the text generation webui inside folders that have !
even if i doubt anyone beside me puts ! inside the folder name because he wants the folder to always be on top of other folders..

**Comment by BigBlueGX** [comment]
I have the same issue and my dir is C:\SD\oobabooga_windows

The automatic installer doesn't work. It gives a Miniconda hook not found error

**Comment by jllllll** [comment]
> > I had the same issue, all my files inside the folder started showing up the error "Miniconda hook not found" and "The system cannot find the path specified"
> > ...after some time.. i tried manually installing miniconda inside my folder. by opening the file that the bat file automatically downloaded.. and... well.. ...it seems that miniconda is unable to install in folders that start with "!" inside. ..my folder was called "!TextGenerationGUI" because i wanted it to always stay on top of other folders.. ..but well.. miniconda is unable to install in folders with "!". so i had rename my folder. ..now, after removing the "!" from the folder name, the problem seems to be solved.
> 
> I think that the installer should warn users that it is impossible to run/install the text generation webui inside folders that have special characters like ! ...even if i doubt anyone beside me puts ! inside the folder name because he wants the folder to always be on top of other folders..

This is pretty good info to know.

**Comment by jllllll** [comment]
> I have the same issue and my dir is C:\SD\oobabooga_windows
> 
> The automatic installer doesn't work. It gives a Miniconda hook not found error

What does the Miniconda installer do if you try to run it manually?

**Comment by BigBlueGX** [comment]
> > I have the same issue and my dir is C:\SD\oobabooga_windows
> > The automatic installer doesn't work. It gives a Miniconda hook not found error
> 
> What does the Miniconda installer do if you try to run it manually?

I tried downloading the automatic installer and ran it just now. I get this error:

 `Conda environment is empty.
Press any key to continue . . .
`

**Comment by jllllll** [comment]
@BigBlueGX I'm not sure why you would be getting `Conda environment is empty. Press any key to continue . . . ` since your log shows that it installed successfully. Does `python.exe` exist in `C:\SD\oobabooga_windows\installer_files\env`?

**Comment by BigBlueGX** [comment]
I see a pythonw.exe in the directory

**Comment by jllllll** [comment]
> @jllllll I see a pythonw.exe in the directory

But not a `python.exe`? That definitely shouldn't be missing. You should check your anti-virus to see if it got removed by it. If not, then you can try re-installing python with this using `cmd_windows.exe`:
```
conda install python=3.10.10
```
This may break the install, not sure. A full re-install may be needed anyway.

**Comment by BigBlueGX** [comment]
@jllllll Disabled my antivirus software and did an automatic installation. You were right. It is now workling

**Comment by PotatoCreator** [comment]
Didn't find a solution from you guys but! This worked for me. Get the one click installer, in start_windows.bat edit the conda download page to "https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe" save and run the bat.

**Comment by ericzhangcn1980** [comment]
> Didn't find a solution from you guys but! This worked for me. Get the one click installer, in start_windows.bat edit the conda download page to "https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe" save and run the bat.

It worked!

**Comment by VincentJGeisler** [comment]
ive tried all the above, even installed conda manually. no joy,  still getting errors, just different one. 

So much broken. 

Downloading Miniconda from https://repo.anaconda.com/miniconda/Miniconda3-py310_23.3.1-0-Linux-x86_64.sh to /mnt/c/Users/admin/Downloads/text-generation-webui-main/text-generation-webui-main/installer_files/miniconda_installer.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 69.7M  100 69.7M    0     0  4248k      0  0:00:16  0:00:16 --:--:-- 3517k
ERROR: File or directory already exists: '/mnt/c/Users/admin/Downloads/text-generation-webui-main/text-generation-webui-main/installer_files/conda'
If you want to update an existing installation, use the -u option.
Miniconda version:
./start_linux.sh: line 41: /mnt/c/Users/admin/Downloads/text-generation-webui-main/text-generation-webui-main/installer_files/conda/bin/conda: No such file or directory
./start_linux.sh: line 46: /mnt/c/Users/admin/Downloads/text-generation-webui-main/text-generation-webui-main/installer_files/conda/bin/conda: No such file or directory
Conda environment is empty.

**Comment by jllllll** [comment]
@VincentJGeisler Are you trying to install with WSL?

If so, I highly recommend using the WSL-specific installer as it contains some additional safeguards against some WSL bugs.
If you prefer to run the WSL installer from within WSL instead of using the `start_wsl.bat` CMD script, then you can execute `wsl.sh` directly with `bash wsl.sh` to have the same effect, just keep in mind that Linux can not process Windows line endings and the file must be converted before execution. This is what the `*_wsl.bat` scripts do:
```sh
sed -i 's/\x0D$//' ./wsl.sh
```
The errors you had with `start_linux.sh` look like the kind that would arise from Windows line endings, so if that is the script you want to run then convert it as well:
```sh
sed -i 's/\x0D$//' ./start_linux.sh
```

**Comment by VincentJGeisler** [comment]
the wsl attempt was pure despairation. It failed under power shell and cmd as well. Cant find miniconda

**Comment by jllllll** [comment]
For the Windows scripts, try to minimize the file path length to where text-generation-webui is stored as Windows has a path length limit. Try moving the webui files to here: `C:\text-generation-webui\`

**Comment by jllllll** [comment]
Another thing to consider with Windows is the user account name. Windows has some restrictions on what the user account can be named, but not nearly enough. Windows will happily allow you to name your user account with names that break a lot of software, such as names with spaces.

I remember one instance from several months ago where someone had a user account named `admin` and had to make a new user account just to run the webui. If I remember correctly, they eventually had to reinstall Windows entirely due to issues arising from that account name. It could have been something else causing the problem, but it is something to consider.

**Comment by VincentJGeisler** [comment]
>  user account named admin

Gah!!!! curse my sysadmin ways!! that's probably it.  The first thing I do on Windows is create a privileged admin account, do what I need to to get things set up then disable it to make sure it doesn't sneak in later not disabled with a blank password (yes that was a thing at one point in time)

Ok I'll blow this away and start over. Probably with ubuntu and just scrap the whole windows thing.  started out with windows because I started off with LM studio. I figured it would be quick and easy, that turned out it doesn't like EsxI running on blades. so 144 xeon cores, half a TB of ram and 2 A6000's and I couldn't get a 7b model to load. 

which is how i ended up here. 

anyways mystery solved, reinstalling, thanks for the heads up

**Comment by SaltyBarnacles** [comment]
> > I had the same issue, all my files inside the folder started showing up the error "Miniconda hook not found" and "The system cannot find the path specified"
> > ...after some time.. i tried manually installing miniconda inside my folder. by opening the file that the bat file automatically downloaded.. and... well.. ...it seems that miniconda is unable to install in folders that start with "!" inside. ..my folder was called "!TextGenerationGUI" because i wanted it to always stay on top of other folders.. ..but well.. miniconda is unable to install in folders with "!". so i had rename my folder. ..now, after removing the "!" from the folder name, the problem seems to be solved.
> 
> I think that the installer should warn users that it is impossible to run/install the text generation webui inside folders that have special characters like ! ...even if i doubt anyone beside me puts ! inside the folder name because he wants the folder to always be on top of other folders..

That solved my problem! Thank you so much,

**Comment by AdlemElvis** [comment]
I had this problem, in my case, I solved it by removing the accents from the "Naração" folder to "Narracao", and it worked.
Tive esse problema, no meu caso, resolvi tirando os acentos da pasta "Narração" para "Narracao", e funcionou.

**Comment by darkwing229** [comment]
I just had the same issue. 

In my case, Oobabooga was looking for Miniconda in my C:/users/miniconda3 folder and there were files missing there, such as conda.exe and others, but I copied the conda files from the installer_files folder to the users folder and the issue got resolved. Hope that helps.
